src/documentation/runners/flink.md

```diff
@@ -6,4 +6,119 @@ redirect_from: /learn/runners/flink/
+The Apache Flink Runner can be used to execute Beam pipelines using [Apache Flink](https://flink.apache.org). When using the Flink Runner you will create a jar file containing your job that can be executed on a regular Flink cluster. It's also possible to execute a Beam pipeline using Flink's local execution mode without setting up a cluster. This is helpful for development and debugging of your pipeline.
+
+The Flink Runner and Flink are suitable for large scale, continuous jobs, and provide:
+
+* A streaming-first runtime that supports both batch processing and data streaming programs
+* A runtime that supports very high throughput and low event latency at the same time
+* Fault-tolerance with *exactly-once* processing guarantees
+* Natural back-pressure in streaming programs
+* Custom memory management for efficient and robust switching between in-memory and out-of-core data processing algorithms
+* Integration with YARN and other components of the Apache Hadoop ecosystem
+
+The [Beam Capability Matrix]({{ site.baseurl }}/documentation/runners/capability-matrix/) documents the supported capabilities of the Flink Runner.
+
+## Flink Runner prerequisites and setup
+
+If you want to use the local execution mode with the Flink runner to don't have to complete any setup.
+
+To use the Flink Runner for executing on a cluster, you have to setup a Flink cluster by following the Flink [setup quickstart](https://ci.apache.org/projects/flink/flink-docs-release-1.1/quickstart/setup_quickstart.html).
+
+For more information, the [Flink Documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.1/) can be helpful.
+
+### Specify your dependency
+
+You must specify your dependency on the Flink Runner.
+
+```java
+<dependency>
+  <groupId>org.apache.beam</groupId>
+  <artifactId>beam-runners-flink_2.10</artifactId>
+  <version>{{ site.release_latest }}</version>
+  <scope>runtime</scope>
+</dependency>
+```
+
+## Executing a pipeline on a Flink cluster
+
+For executing a pipeline on a Flink cluster you need to package your program along will all dependencies in a so-called fat jar and then put that jar into the lib folder of your Flink installation. How you do this depends on your build system but if you follow along the [Beam Quickstart]({{ site.baseurl }}/get-started/quickstart/) these are the steps you have to take:
+
+```
+$ mvn package -Pflink-runner
+$ cp target/word-count-beam-bundled-0.1.jar /path/to/flink/lib/
+```
+The Beam Quickstart Maven project is setup to use the Maven Shade plugin to create a fat jar and the `-Pflink-runner` argument makes sure to include the dependency on the Flink Runner.
+
+For actually running the pipeline you would use the Flink run script like this:
+```
+$ bin/flink run -c org.apache.beam.examples.WordCount lib/word-count-beam-0.1.jar  \
+    --inputFile=/path/to/quickstart/pom.xml  \
+    --output=/tmp/counts \
+    --runner=org.apache.beam.runners.flink.FlinkRunner
```

Comment:
In my recent tests it seemed to be necessary when using `bin/flink` to run but you're right that ideally it should not be necessary. (When running as described in the Quickstart, with the Flink local mode the `o.a.b.r.f` is not necessary.)

Did you run by executing the jar (or via `mvn exec`) or with `bin/flink`?